src/Raven.Server/Documents/AI/AiGen/AbstractChatCompletionClient.cs

```diff
@@ -0,0 +1,458 @@
+﻿using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.IdentityModel.Tokens;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Json;
+using Raven.Client.Util;
+using Sparrow.Json;
+using JsonSerializer = System.Text.Json.JsonSerializer;
+
+
+namespace Raven.Server.Documents.AI.AiGen;
+
+public abstract class AbstractChatCompletionClient : IDisposable
+{
+    private readonly string _model;
+    private readonly HttpClient _client;
+    private readonly string _structuredOutputSchema;
+    private readonly JsonContextPool _contextPool = new JsonContextPool();
```

Comment:
shouldnt' we inject that and use e.g. context pool from the database?

Reply:
Done